### PR TITLE
qli chapter07

### PR DESCRIPTION
**Pull request by qianliok**

### CODE REVIEW

File: chapter07/qli/ex6.pl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+while( <> ){
+  if( /wilma/ && /fred/ ){
```
Comment (MarkWignall) [inline chapter07/qli/ex6.pl:7]: Damm, I forgot about &&.

File: chapter07/qli/ex4.pl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+while( <> ){
+  if( /[A-Z][a-z]/ ){
```
Comment (MarkWignall) [inline chapter07/qli/ex4.pl:7]: While this catches the obvious what about cases where there are capitals at the end of the word not the beginning or other characters i.e. the A-team.

File: chapter07/qli/ex5.pl [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+while( <> ){
+  if( /(\w)\1/ ){
+    print $_;
+  }
```
Comment (pratyp) [inline chapter07/qli/ex5.pl:9]: Seems to have got this wrong like me. The question is to match two non-white space characters, so <code>\S</code> is our friend.

File: chapter07/qli/ex2.pl [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+
+print "Please input some text or files for 'fred' grepping:\n";
+while( <> ){
+  if( /(F|f)red/ ){
```
Comment (glaKris) [inline chapter07/qli/ex2.pl:8]: can save yourself a character here with [fF]

File: chapter07/qli/ex5.pl (hunk 3 above)
Comment (qianliok) [inline chapter07/qli/ex5.pl:9]: oh, I see. `\S` will handles non whitespace characters, which includes various symbols or special characters. Where \w only handles word characters. thanks.

File: chapter07/qli/ex2.pl (hunk 4 above)
Comment (qianliok) [inline chapter07/qli/ex2.pl:8]: Good

**Pull request merged**